### DISCUSSION TITLE
[GENERAL] Mod-Boss announcer - LK death announcer

### DISCUSSION THREAD

**Discussion by chamorrorenzo** [azerothcore/azerothcore-wotlk#3902] [General]
Hello people!

Maybe you encounter the same issue with this mod (great one btw), a few boss wont announce when they are "death"
Im guessing cause they dont really die, but its more like RP events, for example LK, VALITHRIA, Gunship Battle and else

Im trying to "update" this mod, with no much success since im just newbie at it.

What I did was doing this

        if (BossAnnouncerEnable)
        {
            if (boss->GetMap()->IsRaid() && boss->getLevel() > 80 && boss->IsDungeonBoss() && boss->GetEntry() == 36597 || boss->GetEntry() == 38153)

hard to test by myself somehow the boss keeps reseting when the fight start 
So I just spawned the boss at a random raid and kill him and it works, but now its the only boss thats been announce

I know what I did wrong, its calling just for those IDS to be announce

But maybe u can guide me trought it, so I can make this work for the ones that the mod wont call
Maybe if I add another "if" right below?

thanks

https://github.com/azerothcore/mod-boss-announcer

**Comment by chamorrorenzo** [comment]
editing std::string boss_name;

didnt work :/

**Comment by Helias** [comment]
thank you for this thread, but I think that it's best to open an issue in the related module repo.

**Comment by chamorrorenzo** [comment]
> 
> 
> thank you for this thread, but I think that it's best to open an issue in the related module repo.

you are right, but this issue h as been reported there for a year with no response yet.
if someone could guide me where should I start to get this working, that could be enough

**Comment by Helias** [comment]
I can feel your pain.
I didn't work with this module, but I took a look on it.

You have to find a way from the `Creature` class to `recognize` these bosses which are missing.

If you can't, you could hard-code (but I do not recommend this) the entries and make a check by entry in a similar way like:
```C++

enum bossEntries {
  BOSS_ENTRY_1 = 1234,
  BOSS_ENTRY_2 = 1235/
  // and so on
}

void onCreatureKill() {
        if (BossAnnouncerEnable)
        {

          int entries[] = { bossEntries.BOSS_ENTRY_1, bossEntries.BOSS_ENTRY_2 };
          boolean flag_boss = false;

          for ( auto entry : boss->getEntry() ) {
             if (boss->getEntry() == entry) {
                flag_boss = true;
                 break;
              }
          }
          

            if (flag_boss || boss->GetMap()->IsRaid() && boss->getLevel() > 80 && boss->IsDungeonBoss())
            {


```

I don't think that the world bosses are so much if they are like 10 I think you could use this solution, but the best way is `to recognize` the boss from the `Creature` attributes.
